tfx/experimental/distributed_inference/graphdef_experiments/subgraph_partitioning/beam_pipeline_test.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Beam Pipeline."""
+
+import os
+import tensorflow as tf
+import apache_beam as beam
+from apache_beam.testing import test_pipeline
+from apache_beam.testing import util
+
+import graph_partition
+import beam_pipeline
+
+# Since we're using PyFunc to mimic the behavior of a remote op, we need
+# to import it so that we can run the original model. If we don't,
+# the PyFunc ops cannot be loaded into a TF graph.
+import create_complex_graph  # pylint: disable=unused-import
+
+
+# Some info
+def _get_path(folder_name, file_name):
+  return os.path.join(os.path.join(os.path.dirname(__file__), folder_name),
+                      file_name)
+
+example_op_to_filename = {
+    'main': _get_path('complex_graphdefs', 'main_graph.pb'),
+    'remote_op_a': _get_path('complex_graphdefs', 'graph_a.pb'),
+    'remote_op_b': _get_path('complex_graphdefs', 'graph_b.pb')}
+
+example_op_to_outputs = {
+    'main': ['AddN_1'],
+    'remote_op_b': ['Add_1'],
+    'remote_op_a': ['embedding_lookup/Identity']}
+
+example_op_to_remote_op_name_mapping = {
+    'main': {'remote_op_a': {'ids_a': 'ids1'},
+             'remote_op_b': {'ids_b1': 'ids1', 'ids_b2': 'ids2'},
+             'remote_op_a_1': {'ids_a': 'FloorMod_1'},
+             'remote_op_b_1': {'ids_b1': 'FloorMod_1', 'ids_b2': 'FloorMod'}},
+    'remote_op_b': {'remote_op_a': {'ids_a': 'FloorMod'},
+                    'remote_op_a_1': {'ids_a': 'ids_b2'}}}
+
+feed_dicts_main_graph = [
+    {'main': {'import/ids1:0': 3, 'import/ids2:0': 3}},
+    {'main': {'import/ids1:0': 10, 'import/ids2:0': 10}}]
+feed_dicts_graph_b = [
+    {'remote_op_b': {'import/ids_b1:0': 3, 'import/ids_b2:0': 3}},
+    {'remote_op_b': {'import/ids_b1:0': 10, 'import/ids_b2:0': 10}}]
+feed_dicts_graph_a = [
+    {'remote_op_a': {'import/ids_a:0': 3}},
+    {'remote_op_a': {'import/ids_a:0': 10}}]
+example_op_to_feed_dicts = {
+    'main': feed_dicts_main_graph,
+    'remote_op_b': feed_dicts_graph_b,
+    'remote_op_a': feed_dicts_graph_a}
```

Comment:
Are `feed_dicts_graph_a` and `feed_dicts_graph_b` used? This seems to be "cheating" by pre-computing the inputs to the remote op graphs.

Reply:
They're not being used. Removed